storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/ConsumerTask.java

```diff
@@ -220,16 +227,16 @@ public boolean isPartitionAssigned(int partition) {
         return assignedMetaPartitions.contains(partition);
     }
 
-    private void ensureNotClosed() {
-        if (closing) {
-            throw new IllegalStateException("This instance is already closed");
-        }
-    }
-
     public void close() {
         if (!closing) {
-            closing = true;
-            consumer.wakeup();
+            synchronized (assignPartitionsLock) {
+                // Closing should be updated only after acquiring the lock to avoid race in
+                // maybeWaitForPartitionsAssignment() where it waits on assignPartitionsLock. It should not wait
+                // if the closing is already set.
+                closing = true;
```

Comment:
Is it possible two threads call `close()` concurrently?

Reply:
`close()` will not be called concurrently here.